### ISSUE TITLE
PRN output transformations, and what is an appropriate input?

### ISSUE THREAD

**Issue by matth-av** [soullessrobot/LiDAR-HMR#7]
Edit: I have solved my initial issue, please see my questions in the next comment.

**Comment by eidetic-av** [comment]
I was able to find the H36M_J17_NAME, H36M_J17_TO_J14, J24_NAME, J24_TO_J14 variables in another project repo - PointHMR - I guess it is quite a consistent standard. 

I am still working through other issues trying to run the code. Hopefully after these many days of attempting to set up the same environment as your PC I can get it running.

May I suggest that you try and run your train and test code on a different computer or at least a different python virtual environment before publishing it? Then it may save a people's time finding missing dependencies in order to reproduce your tests.

I do not mean to diminish from your contribution at all! Thanks a lot, I am looking forward to running your code! I know python distribution is hard for reproducible research, but it has mainly been solved through projects like conda and docker (along with validating it works on a fresh Linux install).

**Comment by matth-av** [comment]
Ok! I'm able to run your models, I am just focusing on the Pose Regression Network... however the joint positions that are output seem to be in a different 3D space. I have a few questions:
1. Does the `pose` result from the PCTv2_SegReg need some transformation to align with the input point-cloud or the `xyz` output from the model?
2. Is it correct to just input a human point-cloud cropped out from the sloper4d dataset? i.e. this is what my input looks like:
![Screenshot_2024-10-23_12-37-00](https://github.com/user-attachments/assets/ccf59dcb-4d0c-4f23-8e92-03e56e4f2d0a)

This input is not normalised or transformed in any way.
3. When upsampling this data to the required 1024 points, is a simple duplication of vertices ok? or do I need to be creating new unique points.

Here is my current python script in which I'm trying to load the PRN model, loading the pretrained pth that you have uploaded, and then saving the joint positions.

```python
import torch
import torch.nn as nn
import open3d as o3d
import numpy as np
from models.pct_config import pct_config
from models.pct.pct_pose import PCTv2_SegReg

print(f'-- Using PCT config:\n{pct_config}')
print(f'-- Loading model...')

model = PCTv2_SegReg(pct_config).cuda()
print(f'--> Loaded initial model')

checkpoint_path = 'checkpoints/sloper4d/prn_pct.pth'
print(f'-- Loading checkpoint from {checkpoint_path}...')

checkpoint = torch.load(checkpoint_path)
model.load_state_dict(checkpoint['net'])
print(f'--> Loaded checkpoint!')

point_cloud_path = '/mnt/datasets/sloper4d/test_input.pcd'
print(f'-- Loading pointcloud from {point_cloud_path}')

pcd = o3d.io.read_point_cloud(point_cloud_path)

pcd_array = np.asarray(pcd.points)
pcd_array = pcd_array[np.newaxis, :, :]  # Add batch dimension

point_cloud = torch.tensor(pcd_array, dtype=torch.float32).cuda()

print(f'--> Loaded point cloud as tensor!')

print(f'-- Running inference')

model.eval()

with torch.no_grad():
    result = model(point_cloud)

print(f'--> Inference succeeded!')

pose = result['pose'].cpu().numpy()
pose = pose.squeeze(0)

pcd_joints = o3d.geometry.PointCloud()
pcd_joints.points = o3d.utility.Vector3dVector(pose)
output_pcd_path = 'joint_positions.pcd'
o3d.io.write_point_cloud(output_pcd_path, pcd_joints)

```

The input positions from the point cloud look something like:
`[[-0.962, 5.764, 4.541]...]`

And the output for `pose` that I receive from the model looks like:
`[[-2.525, -1.967, 1.143]...]`

Which is not close at all, so I feel like I am missing some normalisation or transformation step.

**Comment by matth-av** [comment]
Here is an image of the input point-cloud above, compared to the output joint positions I am receiving from the PRN in the yellow region.

![Screenshot_2024-10-27_11-01-21](https://github.com/user-attachments/assets/f0658197-79e1-4a36-9fc9-6febf195f559)

**Comment by nhat-m-le** [comment]
@eidetic-av Hi, I am also trying to work with the PRN but couldn't get the right SMPL model. Do you know what we need to do after downloading the SMPL-X models?

**Comment by matth-av** [comment]
@nhatleminh1997 I saw that the path is hard-coded in a few places like [here](https://github.com/soullessrobot/LiDAR-HMR/blob/main/models/pose_mesh_net.py#L157-L158).
So I placed smplx_models directory in the repo root and it looks like this to me:
```
smplx_models
├── smpl
│   └── SMPL_NEUTRAL.pkl
└── smplx
    ├── SMPLX_FEMALE.npz
    ├── SMPLX_FEMALE.pkl
    ├── SMPLX_MALE.npz
    ├── SMPLX_MALE.pkl
    ├── SMPLX_NEUTRAL.npz
    ├── SMPLX_NEUTRAL.pkl
    ├── smplx_npz.zip
    └── version.txt
```
I have tried sampling the input to 1024 points and normalising the point cloud as preprocessing steps but still haven't succeeded getting any kind of expected output. 
Keen to hear if you get any further than I did...

**Comment by soullessrobot** [comment]
Sorry for late reply.
1. the input points are normalized with center of the external cuboid of the input point cloud, which is local points. See in [Dataset file](https://github.com/soullessrobot/LiDAR-HMR/blob/main/datasets/LiDAR_dataset.py) in terms of 'human_points_local'.
2. cropped points are need for the input.
3. Duplication is ok. See in [Dataset file](https://github.com/soullessrobot/LiDAR-HMR/blob/main/datasets/LiDAR_dataset.py)
